group-actors/src/main/java/org/sunbird/service/impl/GroupServiceImpl.java

```diff
@@ -82,15 +85,39 @@ public String createGroup(Group groupObj) throws BaseException {
                 memberService.fetchMembersByGroupIds(Lists.newArrayList(groupId), null);
             dbResGroup.put(JsonKey.MEMBERS, members);
           }
-        }
-        if (CollectionUtils.isEmpty(fields) || !fields.contains(JsonKey.ACTIVITIES)) {
-          dbResGroup.remove(JsonKey.ACTIVITIES);
+          if (fields.contains(JsonKey.ACTIVITIES)) {
+            List<Map<String, Object>> dbResActivities =
+                (List<Map<String, Object>>) dbResGroup.get(JsonKey.ACTIVITIES);
+            if (dbResActivities != null && !dbResActivities.isEmpty()) {
+              addActivityInfoDetails(dbResActivities);
+            }
+          } else {
+            removeActivities(dbResGroup);
+          }
+        } else {
+          removeActivities(dbResGroup);
         }
       }
     }
     return dbResGroup;
   }
 
+  private Object removeActivities(Map<String, Object> dbResGroup) {
+    return dbResGroup.remove(JsonKey.ACTIVITIES);
+  }
+
+  private void addActivityInfoDetails(List<Map<String, Object>> dbResActivities) {
+    Map<String, Map<String, Object>> activityInfoMap =
+        searchActivityService.searchActivity(dbResActivities);
+    for (Map<String, Object> activity : dbResActivities) {
+      if (activityInfoMap.containsKey(activity.get(JsonKey.ID))) {
```

Comment:
@sknirmalkar89 - we should use a combination of activity id and type as the key for activityInfoMap. Two activities of different types may have a same Id and this code will fail in that scenario.